aws-servicecatalog-serviceaction/src/main/java/software/amazon/servicecatalog/serviceaction/CreateHandler.java

```diff
@@ -13,34 +10,27 @@
 
 public class CreateHandler extends BaseHandler<CallbackContext> {
 
-    private static final String CREATE_ERROR_MSG = "Unable to create service action because: %s";
-
     @Override
     public ProgressEvent<ResourceModel, CallbackContext> handleRequest(
-        final AmazonWebServicesClientProxy proxy,
-        final ResourceHandlerRequest<ResourceModel> request,
-        final CallbackContext callbackContext,
-        final Logger logger){
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final Logger logger){
 
         ActionController actionController = ActionController
                 .builder()
                 .logger(logger)
                 .proxy(proxy)
                 .scClient(SCClientBuilder.getClient())
                 .build();
-        final ResourceModel desiredModel = request.getDesiredResourceState();
+        ResourceModel desiredModel = request.getDesiredResourceState();
         final String idempotencyToken = request.getClientRequestToken();
         try {
             final CreateServiceActionResponse response = actionController.createServiceAction(desiredModel, idempotencyToken);
-            final ResourceModel model = ActionController
-                    .buildResourceModelFromServiceActionDetail(response.serviceActionDetail());
-            return ProgressEvent.defaultSuccessHandler(model);
-
-        } catch (InvalidParametersException e) {
-            throw new CfnInvalidRequestException(String.format(CREATE_ERROR_MSG, e.getMessage()), e);
-
-        } catch (LimitExceededException e) {
-            throw new CfnServiceLimitExceededException(ResourceModel.TYPE_NAME, e.getMessage(), e);
+            desiredModel = ActionController.buildResourceModelFromServiceActionDetail(response.serviceActionDetail());
+        } catch (SdkException e) {
+            ExceptionTranslator.translateToCfnException(e);
         }
+        return ProgressEvent.defaultSuccessHandler(desiredModel);
```

Comment:
Why is the return outside of the try catch?

Reply:
Updated code